### DISCUSSION TITLE
How to include magnJ in the post-processing gmsh output?

### DISCUSSION THREAD

**Discussion by Raichuu41** [simnibs/simnibs#201] [8. Simulation workflows] [answered]
In the SimNIBS UI, there is an option to display magnJ (here: https://simnibs.github.io/simnibs/build/html/documentation/gui.html#simulation-options-window). But when using Python with `run_simnibs()`, there is no magnJ in the post-processing available. Is there a place somewhere to set a flag which will include magnJ in the gmsh output? I couldn't find anything on the documentation about it unfortunately.

**Comment by Raichuu41** [comment, marked as answer]
I was searching for a while and just as I posted the question here, I found the solution inside the package code. For future reference, if someone is struggling with the same question:
When creating the session object, this is how you can turn on magnJ in the output:
```python
from simnibs import sim_struct
SESSION = sim_struct.SESSION()
SESSION.fields = f'{SESSION.fields}j'
```

The default currently is `eE` for E and magnE. If you add `j` to the fields string, it will include magnJ in the output. There are a few more letter options. You can find them under `<path-to-package>\simnibs\GUI\simulation_menu.py` in the function at line 117 `def changeFields(self)`. It is a little trial and error to find out which letter stands for what.

**Reply by oulap** [reply]
The field options are documented here:

https://simnibs.github.io/simnibs/build/html/documentation/sim_struct/session.html

**Reply by Raichuu41** [reply]
Ah thank you very much! That would have helped a lot if I found it previously haha.
